src/ray/core_worker/test/fiber_state_test.cc

```diff
@@ -95,6 +95,14 @@ TEST(FiberStateTest, RespectsConcurrencyLimit) {
   fiber_state.Join();
 }
 
+TEST(FiberStateTest, DoubleStopJoin) {
```

Comment:
Besides the unit test, can we also add an e2e test using the repro script mentioned in the GH issue?

Reply:
I don't think this is worth it.
The original repro script is sort of red herring. This test covers the root of the problem.

Reply:
I know this unit test covers the root cause of this issue, but it's still nice to make sure user's workload work well e2e even if, for example, we remove fiber completely in the future(and this unit test will be irrelevant)